dodiscover/context.py

```diff
@@ -127,9 +98,13 @@ def excluded_edges(self) -> nx.Graph:
     def init_graph(self) -> Graph:
         return self._init_graph
 
-    @init_graph.setter
-    def init_graph(self, init_graph: Graph):
-        self._init_graph = init_graph
+    @property
+    def observed_variables(self) -> Set[Column]:
+        return self._variables
+
+    @property
+    def latent_variables(self) -> Set[Column]:
+        return self._latents
 
     def add_state_variable(self, name: str, var: Any) -> None:
         """Add a state variable.
```

Comment:
Is this part okay? 

So the motivation is that during the `LearnSemiMarkovianSkeleton` algorithm, you need to store an additional PAG, which is the "initial skeleton" so that way you can enumerate the PDS sets per node. Think of it as storing a temporary skeleton to query neighbors smartly.

Would it be better to add that capability into the Builder?

cc: @darthtrevino @robertness

Reply:
Otw, I think what would need to occur is inside the `LearnSemiMarkovianSkeleton`, one would need to re-call the builder and build the same Context, but add additional variables (i.e. the intermediate PAG/skeleton)

See here in FCI: https://github.com/py-why/dodiscover/pull/52/files#r981754091

Reply:
Lmk if that needs clarification.

Reply:
Should be okay, but the builder should have a state-variable map it can be initialized with... wip